### ISSUE TITLE
Can't reach nixos.org over ipv6

### ISSUE THREAD

**Issue by Gaius-lafayette** [NixOS/nixos-homepage#1059]
Was having problems updating but couldn't figure out why

```
sudo nix-channel --update
warning: error: unable to download 'https://nixos.org/channels/nixos-22.11': HTTP error 302 (curl error: Timeout was reached); retrying in 266 ms
```

Can't reach over ipv6
```
ping -6 nixos.org -c 4
PING nixos.org(2600:1f18:2489:8201::c8 (2600:1f18:2489:8201::c8)) 56 data bytes

--- nixos.org ping statistics ---
4 packets transmitted, 0 received, 100% packet loss, time 3089ms
```

But I can ping over ipv4
```
ping -4 nixos.org -c 4
PING  (35.231.210.182) 56(84) bytes of data.
64 bytes from 182.210.231.35.bc.googleusercontent.com (35.231.210.182): icmp_seq=1 ttl=58 time=43.2 ms
64 bytes from 182.210.231.35.bc.googleusercontent.com (35.231.210.182): icmp_seq=2 ttl=58 time=42.1 ms
64 bytes from 182.210.231.35.bc.googleusercontent.com (35.231.210.182): icmp_seq=3 ttl=58 time=42.1 ms
64 bytes from 182.210.231.35.bc.googleusercontent.com (35.231.210.182): icmp_seq=4 ttl=58 time=42.1 ms

---  ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3005ms
rtt min/avg/max/mdev = 42.076/42.391/43.205/0.470 ms

```

My ipv6 is working
```
ping -6 google.com -c 4
PING google.com(dfw28s30-in-x0e.1e100.net (2607:f8b0:4000:809::200e)) 56 data bytes
64 bytes from dfw28s30-in-x0e.1e100.net (2607:f8b0:4000:809::200e): icmp_seq=1 ttl=120 time=3.39 ms
64 bytes from dfw28s30-in-x0e.1e100.net (2607:f8b0:4000:809::200e): icmp_seq=2 ttl=120 time=3.67 ms
64 bytes from dfw28s30-in-x0e.1e100.net (2607:f8b0:4000:809::200e): icmp_seq=3 ttl=120 time=3.08 ms
64 bytes from dfw28s30-in-x0e.1e100.net (2607:f8b0:4000:809::200e): icmp_seq=4 ttl=120 time=2.95 ms


--- google.com ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3006ms
rtt min/avg/max/mdev = 2.949/3.271/3.671/0.280 ms
```

 After disabling IPv6 on my interface
```
[root@conceptOS:~]$ sudo nix-channel --update
unpacking channels...

[root@conceptOS:~]$
```
success

I'm using hurricane electric as my IPv6 provider, and there have been issues with people blocking them before (netflix). But if someone could check if ipv6 is working from other providers that would be great for debugging. I can't recall this being an issue in the past so perhaps something automatically added the hurricane electric ISP to a block list or something?

I'm also not sure if this is the right github repository to place this issue. If it's not, can someone please point me in the right direction?

**Comment by Lord-Valen** [comment]
Breadcrumbing https://github.com/NixOS/nixpkgs/issues/233867

**Comment by DamitusThyYeetus123** [comment]
This is still a major issue, and hasn't been fixed in the 2 years since this issue was made. Deploying to a v6-only machine, such as those given by many VPS providers, alongside some mobile networks, is not possible (unless using a v4-capable machine to tunnel through). Alongside this, the AAAA record for nixos.org is entirely gone. Also, nixos.org seems to use AWS global accelerator, which does support dual stack, so implementing this seems trivial. 

EDIT: sorry, this seems to be an issue that has already been fixed. IPv6 was turned off due to another issue (https://github.com/NixOS/infra/issues/873). I'd recommend this be closed to avoid confusion.

**Comment by vcunat** [comment]
No, IPv6 for `nixos.org` does not work right now.  Gandi had issues with ALIAS/ANAME and Netlify [won't provide](https://answers.netlify.com/t/aaaa-ipv6-record-for-apex-domains/11872) a static AAAA.

**Comment by thilobillerbeck** [comment]
We are already on search for a new hosting solution which supports v6. The choice of Netlify comes unfortunately from a time where I have not yet been responsible for the website. 

Due to the need of build mechanisms (CI), preview deployments for PRs and options for a proper root domain DNS setup (meaning A and AAAA records), this takes some time to evaluate. We have some providers in consideration but we did not yet make any decisions.

**Comment by thilobillerbeck** [comment]
Btw. unfortunately most of the popular providers fail at exactly the same point as netlify with a missing fallback for those who want to use Ipv4 and Ipv6 but avoid ALIAS/ANAME records. 

Yes, we could also just set up a VM and put a CDN in front, but I want to consider other options as well to avoid just offloading the work to infra.

**Comment by tgulacsi** [comment]
Hmm, I have gandi.net, and a script upgrades hetzner.gulacsi.eu 's AAAA record regularly on change, and have the www.gulacsi.eu be a CNAME of hetzner.gulacsi.eu. This makes www.gulacsi.eu resolve to the current actual AAAA.
(Poor man's dyndns).

Would this be usable here?

**Comment by vcunat** [comment]
No.  This is not dyndns-like situation.  EDIT: I mean, it would probably somehow partially worked, but I certainly wouldn't do that as a long-term solution.  This kind of CDNs don't rely on anycast, and you don't want to route the whole world's traffic to a single place (bad latencies), so they serve different records to different people.
